web/src/hooks/redux.js

```diff
@@ -102,3 +107,48 @@ export function useSmallLoaderVisible() {
   const windLoading = useSelector(state => state.data.isLoadingWind);
   return gridLoading || solarLoading || windLoading;
 }
+
+export function useCurrentNightTimes() {
+  const { zoneId } = useParams();
+  const zone = useSelector(state => state.data.grid.zones[zoneId]);
+  const datetimeStr = useSelector(state => state.data.grid.datetime);
+  const history = useCurrentZoneHistory();
+
+  return useMemo(() => {
+    if (!zone || !datetimeStr || !history || !history[0]) {
+      return [];
+    }
+    const { latitude, longitude } = getCenteredZoneViewport(zone);
+    const nightTimes = [];
+    let baseDatetime = moment(datetimeStr).startOf('day').toDate();
+    const earliest = history && history[0] && new Date(history[0].stateDatetime);
+    const latest = new Date((history && history[history.length - 1])
+      ? history[history.length - 1].stateDatetime
+      : datetimeStr);
+    do {
+      console.log('got it')
+      // Get last nightTime
+      const nightStart = getSunset(latitude, longitude, baseDatetime);
+      let nightEnd = getSunrise(latitude, longitude, baseDatetime);
+      // Due to some bug in the library, sometimes we get nightStart > nightEnd
+      if (nightStart.getTime() > nightEnd.getTime()) {
+        nightEnd = moment(nightEnd).add(1, 'day').toDate();
+      }
+      // Ignore nights that start after the latest time we have
+      // or that finish before the earliest time we have
+      if (nightStart.getTime() < latest.getTime() && nightEnd.getTime() > earliest.getTime()) {
+        nightTimes.push([nightStart, nightEnd]);
```

Comment:
Why push these dates if you want to ignore them ?

Reply:
Hmm, Oli wrote the comment, but I think the point is that we ignore nights that doesn't match the conditional (so we only push the dates we **don't** ignore). I have rewritten the comment now